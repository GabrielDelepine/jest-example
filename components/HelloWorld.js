import React from 'react'
import { Component, Fragment } from 'react'
import PropTypes from 'prop-types';

export default class HelloWorld extends Component {
  static propTypes = {
    companyName: PropTypes.string,
  }

  render () {
    const { companyName } = this.props

    return (
      <Fragment>
        <h1>Hello World</h1>
        { companyName && <p>company: {companyName}</p> }
      </Fragment>
    )
  }
}
