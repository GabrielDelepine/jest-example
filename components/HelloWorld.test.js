import React from 'react'
import renderer from 'react-test-renderer'
import HelloWorld from './HelloWorld'

test('default props', () => {
  const component = renderer.create(<HelloWorld />)
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot(true)
})

test('companyName props', () => {
  const component = renderer.create(<HelloWorld companyName={'Ubisoft'} />)

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot()
})
