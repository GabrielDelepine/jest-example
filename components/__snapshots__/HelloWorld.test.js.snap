// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`companyName props 1`] = `
Array [
  <h1>
    Hello World
  </h1>,
  <p>
    company: 
    Ubisoft
  </p>,
]
`;

exports[`default props 1`] = `
<h1>
  Hello World
</h1>
`;
